5207 dataverse facet

JayanthyChengan opened the pull request:
## Related Issues

- connects to #5207

scolapasta commented on src/main/java/edu/harvard/iq/dataverse/search/SearchIncludeFragment.java:
```diff
@@ -427,9 +428,9 @@ public void search(boolean onlyDataRelatedToMe) {
-            previewCountbyType.put("dataverses", 0L);
-            previewCountbyType.put("datasets", 0L);
-            previewCountbyType.put("files", 0L);
+            previewCountbyType.put(BundleUtil.getStringFromBundle("dataverses"), 0L);
```
This text isn't actually shown anywhere, correct? I don't think there's a need to localize it.

JayanthyChengan replied:
@scolapasta - This shows in the facets 

![image](https://user-images.githubusercontent.com/12862735/56766966-a7c5b800-6778-11e9-9934-07b3e97b9820.png)

scolapasta replied:
@JayanthyChengan Sorry, what I meant to say is that the text isn't gotten from this previewCountbyType hashmap, as far as I can tell (in the xhtml, I see calls to "{bundle['dataverse.results.types.dataverses']}" and "{bundle['dataverse.results.types.datasets']}". If that's true, then we don't need to localize the internal keys to hashmaps like this. 

(or am I missing something?)

JayanthyChengan replied:
@scolapasta - I replied for this comment to your email.  Please let me know, if I need to forward the email again.  Thanks

scolapasta replied:
@JayanthyChengan Can you go ahead and forward the e-mail again? I couldn't find the reference to this specific comment.

JayanthyChengan replied:
@scolapasta - just forwarded the e-mail.

scolapasta commented:
@JayanthyChengan Could you get the latest from develop and refresh this? I'm hoping to have some more feedback this afternoon.

JayanthyChengan commented:
@scolapasta - updated to latest develop code.

kcondon commented:
@JayanthyChengan I've deployed your code and retrieved the latest bundle files from the /develop branch of https://github.com/GlobalDataverseCommunityConsortium/dataverse-language-packs . I had some trouble initially with deploying the war file if I had defined the language directory first -it could not find the default English bundle. So, I updated the language setting first, deployed your war file, then created the language directory jvm-option, uploaded the languages.zip file and restarted glassfish. The application functions and I'm able to switch between French and English and I see the facet labels display in French but I am seeing many server log errors. Is this a problem with the code or something I've done incorrectly in configuration or deployment? Thanks for any help. 

Kevin

[2019-07-22T14:56:03.448-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=48 _ThreadName=jk-connector(1)] [timeMillis: 1563821763448] [levelValue: 900] [[
  Could not find key "Public" in bundle file: ]]

[2019-07-22T14:56:03.453-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=48 _ThreadName=jk-connector(1)] [timeMillis: 1563821763453] [levelValue: 900] [[
  Could not find key "Restricted" in bundle file: ]]

[2019-07-22T14:56:03.765-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=48 _ThreadName=jk-connector(1)] [timeMillis: 1563821763765] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileAccess" in bundle file: ]]

[2019-07-22T14:58:43.768-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821923768] [levelValue: 900] [[
  Could not find key "Department" in bundle file: ]]

[2019-07-22T14:58:43.771-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821923771] [levelValue: 900] [[
  Could not find key "Journal" in bundle file: ]]

[2019-07-22T14:58:43.775-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821923775] [levelValue: 900] [[
  Could not find key "Laboratory" in bundle file: ]]

[2019-07-22T14:58:44.043-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821924043] [levelValue: 900] [[
  Could not find key "staticSearchFields.dvCategory" in bundle file: ]]

[2019-07-22T14:58:44.046-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821924046] [levelValue: 900] [[
  Could not find key "Root" in bundle file: ]]

[2019-07-22T14:58:44.309-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821924309] [levelValue: 900] [[
  Could not find key "staticSearchFields.metadataSource" in bundle file: ]]

[2019-07-22T14:58:44.312-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821924312] [levelValue: 900] [[
  Could not find key "2019" in bundle file: ]]

[2019-07-22T14:58:44.577-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821924577] [levelValue: 900] [[
  Could not find key "staticSearchFields.publicationDate" in bundle file: ]]

[2019-07-22T14:58:44.639-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821924639] [levelValue: 900] [[
  Could not find key "staticSearchFields.subject_ss" in bundle file: ]]

[2019-07-22T14:58:45.032-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821925032] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileTypeGroupFacet" in bundle file: ]]

[2019-07-22T14:58:45.298-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821925298] [levelValue: 900] [[
  Could not find key "staticSearchFields.dvObjectType" in bundle file: ]]

[2019-07-22T14:58:45.562-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821925562] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileTag" in bundle file: ]]

[2019-07-22T14:58:45.822-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821925822] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileAccess" in bundle file: ]]

[2019-07-22T14:58:46.130-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926130] [levelValue: 900] [[
  Could not find key "Department" in bundle file: ]]

[2019-07-22T14:58:46.133-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926133] [levelValue: 900] [[
  Could not find key "Journal" in bundle file: ]]

[2019-07-22T14:58:46.136-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926136] [levelValue: 900] [[
  Could not find key "Laboratory" in bundle file: ]]

[2019-07-22T14:58:46.399-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926399] [levelValue: 900] [[
  Could not find key "staticSearchFields.dvCategory" in bundle file: ]]

[2019-07-22T14:58:46.402-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926402] [levelValue: 900] [[
  Could not find key "Root" in bundle file: ]]

[2019-07-22T14:58:46.662-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926662] [levelValue: 900] [[
  Could not find key "staticSearchFields.metadataSource" in bundle file: ]]

[2019-07-22T14:58:46.665-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926665] [levelValue: 900] [[
  Could not find key "2019" in bundle file: ]]

[2019-07-22T14:58:46.926-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926926] [levelValue: 900] [[
  Could not find key "staticSearchFields.publicationDate" in bundle file: ]]

[2019-07-22T14:58:46.991-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821926991] [levelValue: 900] [[
  Could not find key "staticSearchFields.subject_ss" in bundle file: ]]

[2019-07-22T14:58:47.126-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821927126] [levelValue: 900] [[
  Could not find key "Tabular Data" in bundle file: ]]

[2019-07-22T14:58:47.129-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821927129] [levelValue: 900] [[
  Could not find key "Image" in bundle file: ]]

[2019-07-22T14:58:47.132-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821927132] [levelValue: 900] [[
  Could not find key "Text" in bundle file: ]]

[2019-07-22T14:58:47.392-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821927392] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileTypeGroupFacet" in bundle file: ]]

[2019-07-22T14:58:47.661-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821927661] [levelValue: 900] [[
  Could not find key "staticSearchFields.dvObjectType" in bundle file: ]]

[2019-07-22T14:58:47.924-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821927924] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileTag" in bundle file: ]]

[2019-07-22T14:58:47.927-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821927927] [levelValue: 900] [[
  Could not find key "Public" in bundle file: ]]

[2019-07-22T14:58:47.930-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821927930] [levelValue: 900] [[
  Could not find key "Restricted" in bundle file: ]]

[2019-07-22T14:58:48.195-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563821928195] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileAccess" in bundle file: ]]

[2019-07-22T14:59:31.202-0400] [glassfish 4.1] [INFO] [] [edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean] [tid: _ThreadID=50 _ThreadName=jk-connector(3)] [timeMillis: 1563821971202] [levelValue: 800] [[
  no user found using kc072219@mailintor.com]]

[2019-07-22T14:59:32.243-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821972243] [levelValue: 900] [[
  Could not find key "Department" in bundle file: ]]

[2019-07-22T14:59:32.246-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821972246] [levelValue: 900] [[
  Could not find key "Journal" in bundle file: ]]

[2019-07-22T14:59:32.249-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821972249] [levelValue: 900] [[
  Could not find key "Laboratory" in bundle file: ]]

[2019-07-22T14:59:32.511-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821972511] [levelValue: 900] [[
  Could not find key "staticSearchFields.dvCategory" in bundle file: ]]

[2019-07-22T14:59:32.514-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821972514] [levelValue: 900] [[
  Could not find key "Root" in bundle file: ]]

[2019-07-22T14:59:32.775-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821972775] [levelValue: 900] [[
  Could not find key "staticSearchFields.metadataSource" in bundle file: ]]

[2019-07-22T14:59:32.779-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821972779] [levelValue: 900] [[
  Could not find key "2019" in bundle file: ]]

[2019-07-22T14:59:33.036-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821973036] [levelValue: 900] [[
  Could not find key "staticSearchFields.publicationDate" in bundle file: ]]

[2019-07-22T14:59:33.297-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821973297] [levelValue: 900] [[
  Could not find key "staticSearchFields.publicationStatus" in bundle file: ]]

[2019-07-22T14:59:33.358-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821973358] [levelValue: 900] [[
  Could not find key "staticSearchFields.subject_ss" in bundle file: ]]

[2019-07-22T14:59:33.746-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821973746] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileTypeGroupFacet" in bundle file: ]]

[2019-07-22T14:59:34.009-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821974009] [levelValue: 900] [[
  Could not find key "staticSearchFields.dvObjectType" in bundle file: ]]

[2019-07-22T14:59:34.499-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821974499] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileTag" in bundle file: ]]

[2019-07-22T14:59:34.757-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821974757] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileAccess" in bundle file: ]]

[2019-07-22T14:59:35.085-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821975085] [levelValue: 900] [[
  Could not find key "Department" in bundle file: ]]

[2019-07-22T14:59:35.088-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821975088] [levelValue: 900] [[
  Could not find key "Journal" in bundle file: ]]

[2019-07-22T14:59:35.091-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821975091] [levelValue: 900] [[
  Could not find key "Laboratory" in bundle file: ]]

[2019-07-22T14:59:35.347-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821975347] [levelValue: 900] [[
  Could not find key "staticSearchFields.dvCategory" in bundle file: ]]

[2019-07-22T14:59:35.350-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821975350] [levelValue: 900] [[
  Could not find key "Root" in bundle file: ]]

[2019-07-22T14:59:35.606-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821975606] [levelValue: 900] [[
  Could not find key "staticSearchFields.metadataSource" in bundle file: ]]

[2019-07-22T14:59:35.609-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821975609] [levelValue: 900] [[
  Could not find key "2019" in bundle file: ]]

[2019-07-22T14:59:35.867-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821975867] [levelValue: 900] [[
  Could not find key "staticSearchFields.publicationDate" in bundle file: ]]

[2019-07-22T14:59:36.132-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821976132] [levelValue: 900] [[
  Could not find key "staticSearchFields.publicationStatus" in bundle file: ]]

[2019-07-22T14:59:36.193-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821976193] [levelValue: 900] [[
  Could not find key "staticSearchFields.subject_ss" in bundle file: ]]

[2019-07-22T14:59:36.326-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821976326] [levelValue: 900] [[
  Could not find key "Tabular Data" in bundle file: ]]

[2019-07-22T14:59:36.329-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821976329] [levelValue: 900] [[
  Could not find key "Image" in bundle file: ]]

[2019-07-22T14:59:36.332-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821976332] [levelValue: 900] [[
  Could not find key "Text" in bundle file: ]]

[2019-07-22T14:59:36.586-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821976586] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileTypeGroupFacet" in bundle file: ]]

[2019-07-22T14:59:36.849-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821976849] [levelValue: 900] [[
  Could not find key "staticSearchFields.dvObjectType" in bundle file: ]]

[2019-07-22T14:59:37.107-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821977107] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileTag" in bundle file: ]]

[2019-07-22T14:59:37.111-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821977111] [levelValue: 900] [[
  Could not find key "Public" in bundle file: ]]

[2019-07-22T14:59:37.114-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821977114] [levelValue: 900] [[
  Could not find key "Restricted" in bundle file: ]]

[2019-07-22T14:59:37.368-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=52 _ThreadName=jk-connector(5)] [timeMillis: 1563821977368] [levelValue: 900] [[
  Could not find key "staticSearchFields.fileAccess" in bundle file: ]]

JayanthyChengan commented:
@kcondon : the exception are thrown due to missing property in bundle.properties.

Please use the property file submitted in this pull request
https://github.com/IQSS/dataverse/blob/0f438ae0de63cda2efe4755a8eb96aafef39caa6/src/main/java/propertyFiles/Bundle.properties

Let me know, if you still notice exceptions.

kcondon commented:
@JayanthyChengan I am still seeing warnings, though not the same or as many. Also, the facet names did not initially appear to be translated to French? Department as a Dataverse facet remains in English as does Subject as a category.

[2019-07-22T15:36:15.481-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563824175481] [levelValue: 900] [[
  Could not find key "Root" in bundle file: ]]

[2019-07-22T15:36:15.802-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563824175802] [levelValue: 900] [[
  Could not find key "2019" in bundle file: ]]

[2019-07-22T15:36:17.598-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563824177598] [levelValue: 900] [[
  Could not find key "1. This is one" in bundle file: ]]

[2019-07-22T15:36:17.603-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563824177603] [levelValue: 900] [[
  Could not find key "2. This is two" in bundle file: ]]

[2019-07-22T15:36:17.609-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563824177609] [levelValue: 900] [[
  Could not find key "3. This is three" in bundle file: ]]

[2019-07-22T15:36:17.619-0400] [glassfish 4.1] [WARNING] [] [edu.harvard.iq.dataverse.util.BundleUtil] [tid: _ThreadID=49 _ThreadName=jk-connector(2)] [timeMillis: 1563824177619] [levelValue: 900] [[
  Could not find key "Documentation" in bundle file: ]]

JayanthyChengan commented:
@kcondon 

**Item 1:**

- Could not find key "2019" in bundle file: ( I don’t think we need to localize year )

- Could not find key "1. This is one" in bundle file:
Could not find key “2. This is two” in bundle file:
Could not find key “3. This is three” in bundle file:

(Not sure where are these coming from, is it related to Facet Category/Facet Label?  , could you please add screenshot.)

Could not find key "Documentation" in bundle file: ( added to bundle.properties)
Could not find key "Département" in bundle file: (it’s related to item 2 )


**Item 2:**
@scolapasta : Noticed in http://localhost:8983/solr/#/collection1/schema?field=dvCategory , the terms in solr index are stored in localized language instead of default language. 
But after indexing, (curl http://localhost:8080/api/admin/index) , it changes all the terms in dvCategory field to default language.
 
When I debug, noticed in dataverse.xhtml , the attribute "itemLabel" in <f:selectItem is internationalized and saved in the solr index with the current language selected instead of default language,   how to handle this?



`<div class="form-group clearfix">
    <label class="col-sm-3 control-label" for="dataverseCategory">
        #{bundle['dataverse.category']} <span class="glyphicon glyphicon-asterisk text-danger" title="#{bundle.requiredField}"/>
        <span class="glyphicon glyphicon-question-sign tooltip-icon"
              data-toggle="tooltip" data-placement="auto right" data-original-title="#{bundle['dataverse.category.title']}"></span>
    </label>
    <div class="col-sm-9 form-col-container">
        <h:selectOneMenu id="dataverseCategory" tabindex="5" styleClass="form-control" value="#{DataversePage.dataverse.dataverseType}">
            <f:selectItem id="dvSelect" itemLabel="#{bundle['dataverse.type.selectTab.top']}" itemValue="" />
            <f:selectItem id="dvDepartment" itemLabel="#{bundle['dataverse.type.selectTab.department']}" itemValue="DEPARTMENT" />
            <f:selectItem id="dvJournals" itemLabel="#{bundle['dataverse.type.selectTab.journals']}" itemValue="JOURNALS" />
            <f:selectItem id="dvLab" itemLabel="#{bundle['dataverse.type.selectTab.laboratory']}" itemValue="LABORATORY" />
            <f:selectItem id="dvOrgInst" itemLabel="#{bundle['dataverse.type.selectTab.organizationsAndInsitutions']}" itemValue="ORGANIZATIONS_INSTITUTIONS" />
            <f:selectItem id="dvResearch" itemLabel="#{bundle['dataverse.type.selectTab.researchers']}" itemValue="RESEARCHERS" />
            <f:selectItem id="dvResearchGrp" itemLabel="#{bundle['dataverse.type.selectTab.researchGroup']}" itemValue="RESEARCH_GROUP" />
            <f:selectItem id="dvResearchProj" itemLabel="#{bundle['dataverse.type.selectTab.researchProjects']}" itemValue="RESEARCH_PROJECTS" />
            <f:selectItem id="dvTeaching" itemLabel="#{bundle['dataverse.type.selectTab.teachingCourses']}" itemValue="TEACHING_COURSES" />
            <f:selectItem id="dvUncategorized" itemLabel="#{bundle['dataverse.type.selectTab.uncategorized']}" itemValue="UNCATEGORIZED" />
        </h:selectOneMenu>
        <p:message for="dataverseCategory" display="text"/>
    </div>
</div>`




**Item 3:**
If the facet category are missing in bundle_fr.properties, it will display in English.
Please add these to the bundle_fr.properties and verify.
("_fr" added to the end of the terms means the translation is in pending)

#FacetCategory - staticSearchFields
staticSearchFields.id=Id
staticSearchFields.dataverseVersionIndexedBy_s=Dataverse Version Indexed By_fr
staticSearchFields.name=Nom
staticSearchFields.description=La description
staticSearchFields.identifier= Identifiant
staticSearchFields.metadataSource=Source de métadonnées
staticSearchFields.isHarvested=Est récolté
staticSearchFields.persistentUrl=URL persistante
staticSearchFields.unf=Unf
staticSearchFields.dvName=Nom du dataverse
staticSearchFields.dvAlias=Alias du dataverse
staticSearchFields.dvAffiliation=Dataverse Affiliation_fr
staticSearchFields.dvDescription=Dataverse Description_fr
staticSearchFields.dvCategory=Catégorie de dataverse
staticSearchFields.dvSubject=Dataverse Sujet
staticSearchFields.subject_ss= Sujet
staticSearchFields.categoryOfDataverse=Category Of Dataverse_fr
staticSearchFields.identifierOfDataverse=Identifier Of Dataverse_fr
staticSearchFields.affiliation_ss=Affiliation
staticSearchFields.fileName=Nom du fichier
staticSearchFields.fileDescription=File Description_fr
staticSearchFields.filePersistentId=File Persistent Id_fr
staticSearchFields.fileType=File Type Searchable_fr
staticSearchFields.fileTypeDisplay=File Type Friendly_fr
staticSearchFields.fileContentType=File Content Type_fr
staticSearchFields.fileTypeGroupFacet=Type de fichier
staticSearchFields.fileSizeInBytes=File Size In Bytes_fr
staticSearchFields.fileMd5=File Md5_fr
staticSearchFields.fileChecksumType=File Checksum Type_fr
staticSearchFields.fileChecksumValue=File Checksum Value_fr
staticSearchFields.fileNameWithoutExtension=Filename Without Extension_fr
staticSearchFields.fileTag=Libellé de fichier
staticSearchFields.fileTags=File Tag Searchable_fr
staticSearchFields.tabularDataTag=Tabdata Tag_fr
staticSearchFields.fileAccess= Accès
staticSearchFields.subtreePaths=Subtree_fr
staticSearchFields.score= Pertinence
staticSearchFields.dvObjectType=Type
staticSearchFields.nameSort=Name Sort_fr
staticSearchFields.publicationDate=Année de publication
staticSearchFields.dateSort=Release Or Create Date_fr
staticSearchFields.definitionPointDocId=Definition Point_fr
staticSearchFields.definitionPointDvObjectId=Definition Point Dvobject Id_fr
staticSearchFields.discoverableBy=Discoverable By_fr
staticSearchFields.publicationStatus=Statut de publication
staticSearchFields.entityId=Entity Id_fr
staticSearchFields.parentName=Parent Name_fr
staticSearchFields.parentId=Parent Id_fr
staticSearchFields.parentIdentifier=Parent Identifier_fr
staticSearchFields.parentCitation=Parent Citation_fr
staticSearchFields.dsDescriptionValue=Dataset Description_fr
staticSearchFields.citation=Citation de l'ensemble de données
staticSearchFields.citationHtml=Dataset Citation Html_fr
staticSearchFields.deaccessionReason=Dataset Deaccession Reason_fr
staticSearchFields.dsPublicationDate=Dataset Publication Date_fr
staticSearchFields.dsPersistentId=Identifiant pérenne de l'ensemble de données
staticSearchFields.datasetVersionId=Dataset Version Id_fr
staticSearchFields.variableName=Nom de la variable
staticSearchFields.variableLabel=Libellé de variable
staticSearchFields._text_=Full Text_fr

#dataverse category - Facet Labels
Researcher=Chercheur
Research\u0020Project=Projet de recherche
Journal=Revue
Organization\u0020or\u0020Institution=Organisation ou établissement
Teaching\u0020Course=Cours
Research\u0020Group=Groupe de recherche
Laboratory=Laboratoire
Department=Département
Uncategorized=Sans catégorie 

#filetype - Facet Labels
Document=Document
Text=Texte
Tabular\u0020Data=Données tabulaires
Data=Données
FITS=FITS
Shape=Formes
ZIP=ZIP
Image=Image
Network\u0020Data=Données en réseau
Unknown=Inconnu
Documentation=Documentation
Code=Code
Archive=Dépôt


#access - Facet Labels
Public=Public
Restricted=Restreint

 

**Item 5:**
The display value for FacetCategory(bold titles ) and Facet labels are fetched either from metadatablock property files or bundle.properties .



**Item 6:**
Removed “ZIP” and added Code and Archive to bundle.properties

scolapasta commented:
For item 2, the issue isn't that it is directly using the label; it is in fact storing the value in the db. But the undexer (indexServiceBean:161) calls a method on Dataverse called  (Dataverse:137):
'''dataverse.getIndexableCategoryName()'''
which in turn calls (Dataverse:112):
'''getFriendlyCategoryName()'''
and that gets the name from the bundle. 

This needs to be fixed either as part of this PR or separately.

The same issue with file types (and possible other CV type fields?):
#5473

scolapasta commented:
The filetypes are gotten in this method (FileUtil:277):
``` public static String getFacetFileType(DataFile dataFile)```

JayanthyChengan commented:
@scolapasta - Thanks for the clarification for item 2.  

I think we can fix this by adding a new method in BundleUtil.java , to return default (English) value for dataverse Category and File type.  Do you prefer separate PR?

scolapasta commented:
@JayanthyChengan I think that solution would work, yes. Could you add as a separate PR? (and could you check other fields, like subject, to make sure they are not affected?)

kcondon commented:
@JayanthyChengan 
For 1. log file warnings, there are none when I deploy develop with the same db, index, and languages and bundles defined. I have commented inline to the warnings above about where the strings are referenced. I am not sure how else I can help but we need to resolve those.
For 4. page load performance, it is really bad when languages are defined, even on develop. It is almost unusable on my test box. I'm not sure what the problem is here @scolapasta any ideas? When I disable languages, even on your branch, performance is fine.
For 2. index in multiple languages, I'll wait for new code.
For 3. and 5. update Bundles .fr, I can do that but what is the process, should it be checked in somewhere?
For 6. thanks. any thoughts on audio and video? @mheppler thought you might also need to update from develop due to some recent changes.

JayanthyChengan commented:
> @JayanthyChengan I think that solution would work, yes. Could you add as a separate PR? (and could you check other fields, like subject, to make sure they are not affected?)

@scolapasta - submitted PR https://github.com/IQSS/dataverse/pull/6051 , I also tested for controlled values (Subject) , it works fine

kcondon commented:
@JayanthyChengan Would you mind refreshing this branch from develop since we have just merged your other branch? Thanks.

JayanthyChengan commented:
@kcondon - merged the latest develop branch.

JayanthyChengan commented:
@kcondon  @pdurbin 

In SearchServiceBean.java, the staticsearchfields declared [here](https://github.com/scholarsportal/dataverse/blob/5207-dataverse-Facet/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java#L343) as

Field[] staticSearchFields = searchFieldsObject.getClass().getDeclaredFields();

and processed in a for loop at Line no: 673

So, I added all the fields mentioned in SearchFields.java to bundle.properties

pdurbin commented:
> Field[] staticSearchFields = searchFieldsObject.getClass().getDeclaredFields();

@JayanthyChengan ah yes, I remember writing this Javadoc comment above the code you found:

"todo: refactor SearchFields to a hashmap (or something? put in database? internationalize?) to avoid the crazy reflection and string manipulation below"

Are you removing the need for that crazy reflection? Will be be able to delete that reflection code? At the time I was advocating a bit for storing more search fields in the database, mostly because I thought working on "Don't Hardcode Any Facets" at #2039 would be nice to do some day. Right now you can only adjust the order of the dynamic, database-driven metadata block facets.

JayanthyChengan commented:
@pdurbin - I don't intend on working on #2039, but even if I attempt to do it, I am not very clear with the complete scope or logic associated with searchfields and database.  Sorry!

scolapasta commented:
@JayanthyChengan let's leave #2039 out of scope for this. I think what is being said is that all these static search fields are not facetable, so don't need to be localized. Let's remove the ones that are not facets (because we'll never going to add a facet on something like "identifier" or "file name"). You should be able to determine from the code which ones are facets, but let me/us know if you want help determining that.

The other thing that was being seen about the entries in the log file - it seems (and again @kcondon feel free to correct or add info) that the code is attempting to localize all of the facets and not just for those that are controlled vocabulary. For the ones that are not, the values are user entered so will usually not be found in bundles? I think we decided that user entered values were out of scope for this issue, so if that's the case, can you look at the code and make sure this is only for facet category labels, and for facet values that come from either controlled vocab or enums in the code?

pdurbin commented:
@JayanthyChengan no problem. I was just telling @kcondon that I like what you've done in this pull request, how you're pulling strings like "Dataverse Category" out of the bundle instead of my old crazy way of transforming DATAVERSE_CATEGORY to the a friendlier string by replacing the underscore with a space and by changing it from upper case. It looks like we still need the reflection to get a list of fields that are declared in SearchFields.java. I still think that some day these should probably go in the database somewhere. Some day. Not today, not for this pull request. 😄

JayanthyChengan commented:
@scolapasta 
1. To remove staticsearchfields that are not facetable from bundle
                    I looked at the code IndexServiceBean.java and planning to add only fields that are used in IndexServiceBean.java into bundle.properties, hoping those are the Facetable fields.  
But found SearchFields.IDENTIFIER , SearchFields.FILE_NAME - are those facetable?  

2. Yes, we are not localizing the user entered values.  Added try/catch to handle exception.

pdurbin commented:
>  But found SearchFields.IDENTIFIER , SearchFields.FILE_NAME - are those facetable?

@JayanthyChengan no, neither of these fields are facetable.

Generally, unique values don't make good facets. For example, every dataset description is different. Facets are best when then accumulate the same value such as the same dataset author.

JayanthyChengan commented:
Thanks @pdurbin .  As @scolapasta mentioned in the last comment, I added only the staticSearchFields that are used for facets into bundle.properties ( referred  solrQuery.addFacetField() statements from SearchServiceBean.java ) , please let me know if I am missing any static fields that are facetable.  Thanks

pdurbin commented:
@JayanthyChengan does that mean you're ready for me to drag this back to code review?

JayanthyChengan commented:
@pdurbin - yes, please move it to code review/ . Thanks
